libtsuba/src/AsyncOpGroup.cpp

```diff
@@ -0,0 +1,51 @@
+#include "tsuba/AsyncOpGroup.h"
+
+bool
+tsuba::AsyncOpGroup::FinishOne() {
+  auto op_it = pending_ops_.begin();
+  if (op_it == pending_ops_.end()) {
+    return false;
+  }
+  auto res = op_it->result.get();
+  if (!res) {
+    KATANA_LOG_DEBUG(
+        "async op for {} returned {}", op_it->location, res.error());
+    errors_++;
+    last_error_ = res.error();
+  } else {
+    res = op_it->on_complete();
+    if (!res) {
+      KATANA_LOG_DEBUG(
+          "complete cb for async op for {} returned {}", op_it->location,
+          res.error());
+    }
+  }
+  pending_ops_.erase(op_it);
+  return true;
+}
+
+katana::Result<void>
+tsuba::AsyncOpGroup::Finish() {
+  while (FinishOne()) {
+    // Wait for all ops
+  }
+
```

Comment:
Is it important for this busy waiting to happen at full speed?  Might we consider the `_mm_pause();` intrinsic?  Or a `yield`?

Reply:
`FinishOne` blocks until the next future it ready so this waiting isn't that busy.